src/components/Calculator.js

```diff
@@ -1,33 +1,173 @@
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(e) {
+    this.setState((previousState) => calculate(previousState, e.target.innerText));
+  }
+
   render() {
+    let val = 0;
+    const { total, next } = this.state;
+    if (next) {
+      val = next;
+    } else if (total) {
+      val = total;
+    }
+
     return (
-      <div className="main-wrap">
-        <div className="container">
-          <div className="operator display">0</div>
-          <div className="operator AC">AC</div>
-          <div className="operator operations">+/-</div>
-          <div className="operator percentage">%</div>
-          <div className="operator divide operation">÷</div>
-          <div className="operator seven">7</div>
-          <div className="operator eight">8</div>
-          <div className="operator nine">9</div>
-          <div className="operator multiply operation">×</div>
-          <div className="operator four">4</div>
-          <div className="operator five">5</div>
-          <div className="operator six">6</div>
-          <div className="operator minus operation">-</div>
-          <div className="operator one">1</div>
-          <div className="operator two">2</div>
-          <div className="operator three">3</div>
-          <div className="operator plus operation">+</div>
-          <div className="operator zero">0</div>
-          <div className="operator period">.</div>
-          <div className="operator equal operation">=</div>
-        </div>
+      <div className="wrapper">
+        <input
+          className="operator display"
+          placeholder="0"
+          value={val}
+          readOnly
+        />
+        <button
+          type="button"
+          className="operator AC"
+          onClick={this.handleClick}
+        >
+          AC
+        </button>
+        <button
+          type="button"
+          className="operator operations"
+          onClick={this.handleClick}
+        >
+          +/-
+        </button>
+        <button
+          type="button"
+          className="operator percentage"
+          onClick={this.handleClick}
+        >
+          %
+        </button>
+        <button
+          type="button"
+          className="operator divide operation"
+          onClick={this.handleClick}
+        >
+          ÷
+        </button>
+        <button
+          type="button"
+          className="operator seven"
+          onClick={this.handleClick}
+        >
+          7
+        </button>
+        <button
+          type="button"
+          className="operator eight"
+          onClick={this.handleClick}
+        >
+          8
+        </button>
+        <button
+          type="button"
+          className="operator nine"
+          onClick={this.handleClick}
+        >
+          9
+        </button>
+        <button
+          type="button"
+          className="operator multiply operation"
+          onClick={this.handleClick}
+        >
+          ×
```

Comment:
- [ ] @Victoria129 great work on your application, however every time I try some multiplication, when I click the equal button the calculation disappears, check the loom video demonstrating it , kindly fix it https://loom.com/share/b2ba29c307bd4bd98eb7968fc04b07e2